src/Build/Logging/BinaryLogger/ProjectImportsCollector.cs

```diff
@@ -140,11 +142,13 @@ bool TryAddFile()
             {
                 try
                 {
-                    addFileWorker(filePath);
+                    addFileWorker(filePath); 
                     return true;
                 }
-                catch
-                { }
+                catch (Exception e)
```

Comment:
If we attempt to catch IO-specific expectations, I would narrow it down with this condition (see the example here:  https://github.com/dotnet/msbuild/blob/b8e5cce93e2e664eaf470c0fed9f5b84b8591e1e/src/Tasks/GetAssemblyIdentity.cs#L71)